### PR TITLE
Fixes #557

### LINKED ISSUE #557: CONDITIONAL_FORWARDING[...] env variables not respected
This is a... 

- [ ] Request for a new or modified feature
- [x] Issue trying to run the docker image
- [ ] Issue trying to build / test / develop the docker image

## Description

CONDITIONAL_FORWARDING environment variables passed via docker run or docker-compose are visible in admin ui but not respected.

## Expected Behavior

Conditional forwarding environment variables should be respected

## Actual Behavior

queries to conditional forwarding domains are blocked by pi-hole/not forwarded until settings are manually saved (not modified) in admin UI

## Steps to Reproduce and debugging done

1. delete/disable etc-pihole volume/bind mount if exists
2. run container with conditional forwarding parameters
3. flush test device's dns cache
4. query for host in conditional forwarding domain, get NXDOMAIN response
5. in pi-hole admin ui, navigate to settings>dns and press "save" button
6. query again, get valid response.  

## Debug steps I have tried

- [x] I have tried destroying my container instance, pulling the newest image version, and re-creating a new container
- [x] I have tried running the nearly stock `docker run` example in the readme (removing any customizations I added)
- [x] I have tried running without my volume data mounts to eliminate volumes as the cause
- [x] I have searched this repository for existing issues and pull requests that look similar 

## Your Environment 

* Docker Host Operating System and OS Version: macOS (latest)
* Docker Version: Docker Desktop for Mac (latest)
* Hardware architecture:  x64

### PR DESCRIPTION
**Pull request by ahuffman**
* Refactored the prepare_configs function in bash_functions.sh
* Now able to set settings via the UI or ENV Variables - ENV Variables override when restarted
* Added ENV Variables for ADMIN_EMAIL, TEMPERATUREUNIT, and WEBUIBOXEDLAYOUT
    - Created setup_temp_unit function in bash_functions.sh
    - Created setup_ui_layout function in bash_functions.sh
    - Created setup_admin_email function in bash_functions.sh
* Updated README.md
    - Added docs on new ENV Variables
    - Updated True/False settings docs so they were quoted, otherwise misleading to k8s users and not "true" booleans

## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [? ] My code follows the code style of this project.
- [x ] My change requires a change to the documentation.
- [x ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ahuffman** [comment]
# Testing performed
* Tested code updates with private build of container image.
* Tested setting values via the UI which overrode an ENV Var, killed the pod and ENV Var restored.
* Tested setting unset values via the UI (not set via ENV var) restarted pod and they "kept"
* Tested overriding the set values via UI and ENV Vars overrode as expected behavior

**Comment by nathang21** [comment]
Is there any reason why this hasn't been merged yet? 
Might need to be rebased for 5.0.

**Comment by diginc** [comment]
Woops I forgot to click submit on my review

File: bash_functions.sh [hunk 1]
```diff
@@ -60,27 +60,11 @@ prepare_configs() {
...
     # If the setup variable file exists,
     if [[ -e "${setupVars}" ]]; then
-        # update the variables in the file
-        local USERWEBPASSWORD="${WEBPASSWORD}"
-        . "${setupVars}"
-        # Stash and pop the user password to avoid setting the password to the hashed setupVar variable
-        WEBPASSWORD="${USERWEBPASSWORD}"
-        # Clean up old before re-writing the required setupVars
-        sed -i.update.bak '/PIHOLE_INTERFACE/d;/IPV4_ADDRESS/d;/IPV6_ADDRESS/d;/QUERY_LOGGING/d;/INSTALL_WEB_SERVER/d;/INSTALL_WEB_INTERFACE/d;/LIGHTTPD_ENABLED/d;' "${setupVars}"
+        cp -f "${setupVars}" "${setupVars}.update.bak"
```
[the remark this answers is not in the document]
Comment (ahuffman) [reply bash_functions.sh:66]: Out in the main start.sh we use the bash_functions.sh prepare_configs function to ensure req'd settings are there, and then we copy the existing config.  Once the prepare_configs is complete we apply settings changes from the environment vars.  So if you apply changes via the UI they stay, unless overridden by an environment var.  The environment vars will always "win" VS. UI changed settings on a container restart. 

I'm personally running this on kubernetes 1.18 on a rpi4 cluster so it's fairly easy to test.  If you/anyone would like to triple verify, the link to my image is in my previous comments.

**Pull request merged**